### PR TITLE
Add method to force a refresh from a cached function

### LINKED ISSUE #18: Add `cache.function` mechanism to temporarily skip cache
Cache is good to avoid slow requests when the data isn't immediately useful, but sometimes you need fresh data.

If you use `cache.get`/`set` you can already implement your own logic, but with `cache.function`, you can't skip the cache.

A "skip cache" mechanism would call the getter function **and** store the result in the cache as usual.

Ideas:

```js
const fn = cache.function(heavyFn);

const value = true or false ? await fn.fresh() : await fn();
```

or better:

```js
const fn = cache.function(heavyFn, {
	needsFreshData: () => true or false
});

const value = await fn();
```

### PR DESCRIPTION
**Pull request by cheap-glitch**
Resolves #18

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
+import chromeP from 'webext-polyfill-kinda';
 import microMemoize from 'micro-memoize';
 import {isBackgroundPage} from 'webext-detect-page';
 import toMilliseconds, {TimeDescriptor} from '@sindresorhus/to-milliseconds';
-import chromeP from 'webext-polyfill-kinda';
```
Comment (cheap-glitch) [inline index.ts:4]: Lint :slightly_smiling_face:

File: index.test-d.ts [hunk 2]
```diff
@@ -18,14 +18,10 @@ expectAssignable<Promise<Record<string, any[]>>>(cache.set('key', {wow: [true, '
 expectAssignable<Promise<number>>(cache.set('key', 1, {days: 1}));
 
 const cachedPower = cache.function(async (n: number) => n ** 1000);
-expectType<(n: number) => Promise<number>>(cachedPower);
+expectType<((n: number) => Promise<number>) & {fresh: (n: number) => Promise<number>}>(cachedPower);
 expectType<number>(await cachedPower(1));
 
-expectType<(n: string) => Promise<number>>(
-	cache.function(async (n: string) => Number(n)),
-);
```
Comment (cheap-glitch) [inline index.test-d.ts:26]: Seems like this test case was duplicated?

**Review by fregante — APPROVED** [review]
LGTM

File: index.ts [hunk 3]
```diff
@@ -154,7 +154,13 @@ function function_<
 		}
 
 		return cachedItem.data;
-	}) as Getter);
+	}) as Getter), {
+		fresh: (async (...args: Arguments) => {
+			const userKey = cacheKey ? cacheKey(args) : (args[0] as string);
```
Comment (fregante) [inline index.ts:159]: I'm not sure why I didn't set this as a default parameter in `function_`:

```js
{
		cacheKey = ([firstArg]) => firstArg,
		etc
}
```

**Pull request merged**